### PR TITLE
Ensure json documents are valid and they end with an empty EOL

### PR DESCRIPTION
**Pull request by serroba**
## What?

Add JSON validator
## Why?

So we ensure our JSON documents are valid and they all end up with an empty EOL

## Testing/Proof

CI will do the work

### CODE REVIEW

File: .github/workflows/json-validator.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install jq
+        run: sudo apt-get install jq
+
+      - name: Validate JSON
+        run: |
+          for file in $(find . -type f -name '*.json'); do
```
Comment (ch-evergen) [inline .github/workflows/json-validator.yaml:24]: this should exclude `.venv` from the `find` - might not be a problem in CI but better for running locally

File: .github/workflows/json-validator.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            jq . "$file" >/dev/null || { echo "Invalid JSON: $file"; exit 1; }
+            last_char=$(tail -c 1 "$file")
```
Comment (ch-evergen) [inline .github/workflows/json-validator.yaml:26]: Not sure if this works consistently - it errors out on every file for me, even those that clearly have newlines at the end.

StackOverflow suggests something like this to detect files without a concluding newline:
`if [[ $(tail -c1 "$file" | wc -l) -eq 0 ]]`

File: .github/workflows/json-validator.yaml (hunk 1 above)
Comment (serroba) [reply .github/workflows/json-validator.yaml:24]: oh! good point!

File: .github/workflows/json-validator.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        uses: actions/checkout@v3
+
+      - name: Install jq
+        run: sudo apt-get install jq
+
+      - name: Validate JSON
+        run: |
+          for file in $(find . -type f -name '*.json' | grep -v -e '/venv/' -e '/.venv/'); do
+            jq . "$file" >/dev/null || { echo "Invalid JSON: $file"; exit 1; }
+            if [[ $(tail -c1 "$file" | wc -l) -eq 0 ]]
```
Comment (ch-evergen) [inline .github/workflows/json-validator.yaml:26]: this needs a `; then`

Comment (serroba) [reply .github/workflows/json-validator.yaml:26]: I wonder how we could be testing this better. Thanks for noticing it though

**Review by ch-evergen — APPROVED** [review]

**Pull request merged**